lib/ApparelSorter.js

```diff
@@ -104,91 +105,75 @@ var regexes = [
     {regex: /^XXL.*$/i, normalized: '2XL'},
     {regex: /^2xt$/i, normalized: '2XL'},
     {regex: /^2xlt$/i, normalized: '2XL'},
-    {regex: /^2x.*$/i, normalized: '2XL'},
-    {regex: /^3x*$/i, normalized: '3XL'},
+    {regex: /^3x$/i, normalized: '3XL'},
     {regex: /^3xl*$/i, normalized: '3XL'},
 	{regex: /^XXXL.*$/i, normalized: '3XL'},
     {regex: /^3tg.*$/i, normalized: '3XL'},
 	{regex: /^tttg.*$/i, normalized: '3XL'},
     {regex: /^3xlt$/i, normalized: '3XL'},
     {regex: /^3xt$/i, normalized: '3XL'},
-    {regex: /^3x.*$/i, normalized: '3XL'},
-    {regex: /^4x*$/i, normalized: '4XL'},
+    {regex: /^4x$/i, normalized: '4XL'},
     {regex: /^4xl*$/i, normalized: '4XL'},
     {regex: /^XXXXL.*$/i, normalized: '4XL'},
     {regex: /^4xlt.*$/i, normalized: '4XL'},
     {regex: /^4xt.*$/i, normalized: '4XL'},
-    {regex: /^4x.*$/i, normalized: '4XL'},
-    {regex: /^5x*$/i, normalized: '5XL'},
+    {regex: /^5x$/i, normalized: '5XL'},
     {regex: /^5xl*$/i, normalized: '5XL'},
 	{regex: /^XXXXXL.*$/i, normalized: '5XL'},
     {regex: /^5xlt.*$/i, normalized: '5XL'},
-    {regex: /^5x.*$/i, normalized: '5XL'},
-    {regex: /^6x*$/i, normalized: '6XL'},
+    {regex: /^6x$/i, normalized: '6XL'},
     {regex: /^6xl*$/i, normalized: '6XL'},
 	{regex: /^XXXXXXL.*$/i, normalized: '6XL'},
     {regex: /^6xlt.*$/i, normalized: '6XL'},
-    {regex: /^6x.*$/i, normalized: '6XL'},
-    {regex: /^7x*$/i, normalized: '7XL'},
+    {regex: /^7x$/i, normalized: '7XL'},
     {regex: /^7xl*$/i, normalized: '7XL'},
 	{regex: /^XXXXXXXL.*$/i, normalized: '7XL'},
     {regex: /^7xlt.*$/i, normalized: '7XL'},
-    {regex: /^7x.*$/i, normalized: '7XL'},
```

Comment:
Some of these removed regexes were causing errant issues. For instance, this regex returns true when a size is a pure int, ie, 4. This came down to the way the regexes are written, with a period after '7x' for instance. In the future, we may want to explore getting away from a 'list of regexes' model.